### PR TITLE
Fixed #31720 -- Defined default output_field of BoolAnd() and BoolOr() aggregate functions and added examples to docs.

### PR DESCRIPTION
**Pull request by davidchorpash**
https://code.djangoproject.com/ticket/31720

Added defaults to BoolOr and BoolAnd classes and made documentation improvements by adding examples.

### CODE REVIEW

File: docs/ref/contrib/postgres/aggregates.txt [hunk 1]
```diff
@@ -75,6 +75,15 @@ General-purpose aggregation functions
...
     null or if there are no values, otherwise ``False`` .
 
+    Example:
+    	
+	Assume ``some_boolean_field`` is a non-nullable BooleanField and
+	there exists at least one instance of ``SomeModel`` where
+	``some_boolean_field = False``:
+        
+	>>> SomeModel.objects.aggregate(booland=BoolAnd('some_bool_field')) 
+	{booland: False}
```
Comment (charettes) [inline docs/ref/contrib/postgres/aggregates.txt:85]:
```suggestion
	{'booland': False}
```

File: docs/ref/contrib/postgres/aggregates.txt [hunk 2]
```diff
@@ -83,6 +92,15 @@ General-purpose aggregation functions
...
     values are null or if there are no values, otherwise ``False``.
 
+    Example:
+    	
+	Assume ``some_boolean_field`` is a non-nullable BooleanField and
+	there exists at least one instance of ``SomeModel`` where
+	``some_boolean_field = True``:
+        
+	>>> SomeModel.objects.aggregate(boolor=BoolOr('some_bool_field')) 
+	{boolor: True}
```
Comment (charettes) [inline docs/ref/contrib/postgres/aggregates.txt:102]:
```suggestion
	{'boolor': True}
```

File: docs/ref/contrib/postgres/aggregates.txt [hunk 3]
```diff
@@ -75,6 +75,28 @@ General-purpose aggregation functions
...
+    BoolAnd can be used with :ref:`Q objects<complex-lookups-with-q>`.
+
+    Example:
+
+	Assume ``some_int_field`` is a non-nullable IntegerField and there
+	exists at least one instance of ``SomeModel`` where ``some_int_field``
+	is a value of 2 or less:
+
+	>>> SomeModel.objects.aggregate(booland=BoolAnd(Q('some_int_field__gt=2')))
+	{'booland': False}
```
Comment (felixxm) [inline docs/ref/contrib/postgres/aggregates.txt:97]: I would change examples to more explicit (like in [Database functions docs](https://docs.djangoproject.com/en/3.0/ref/models/database-functions/)):
 ```suggestion
    Usage example::
        
        class Comment(models.Model):
            body = models.TextField()
            published = models.BooleanField()
            rank = models.IntegerField()
        
        >>> from django.contrib.postgres.aggregates import BitAnd
        >>> Comment.objects.aggregate(boolor=BoolAnd('published')
        {'boolor': False}
        >>> Comment.objects.aggregate(boolor=BoolAnd(Q(rank__lt=100)))
        {'boolor': True}
```

File: docs/ref/contrib/postgres/aggregates.txt [hunk 4]
```diff
@@ -83,6 +105,26 @@ General-purpose aggregation functions
...
+    BoolOr can be used with :ref:`Q objects<complex-lookups-with-q>`.
+    
+    Example:
+
+	Assume ``some_int_field`` is a non-nullable IntegerField and there
+	exists at least one instance of ``SomeModel`` where ``some_int_field``
+	is a value greater than 2:
+
+	>>> SomeModel.objects.aggregate(boolor=BoolOr(Q('some_int_field__gt=2')))
+	{'boolor': True}
```
Comment (felixxm) [inline docs/ref/contrib/postgres/aggregates.txt:126]:
```suggestion
    Usage example::
        
        class Comment(models.Model):
            body = models.TextField()
            published = models.BooleanField()
            rank = models.IntegerField()
        
        >>> from django.contrib.postgres.aggregates import BitOr
        >>> Comment.objects.aggregate(boolor=BoolOr('published')
        {'boolor': True}
        >>> Comment.objects.aggregate(boolor=BoolOr(Q(rank__gt=2)))
        {'boolor': False}
```

Comment (felixxm) [reply docs/ref/contrib/postgres/aggregates.txt:126]: I've prepared edits, so I can push them to you branch, if you agree with my suggestions.

Comment (davidchorpash) [reply docs/ref/contrib/postgres/aggregates.txt:126]: @felixxm I like those suggestions!  They are bit more concise and to the point, in my opinion.

File: docs/ref/contrib/postgres/aggregates.txt (hunk 3 above)
Comment (davidchorpash) [reply docs/ref/contrib/postgres/aggregates.txt:97]: Thanks for the link.  I'll use that as a reference from now on.

**Comment by felixxm** [comment]
@davidchorpash Thanks :+1: Welcome aboard :boat: 

I pushed edits.

**Pull request merged**